### PR TITLE
Update exception message for groupBy() and indexBy() missing path

### PR DESCRIPTION
**Pull request by othercorey**
This also adds a missing test for callbacks that return null values.

### CODE REVIEW

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -269,10 +269,11 @@ public function groupBy($path): CollectionInterface
...
             $pathValue = $callback($value);
             if ($pathValue === null) {
-                throw new InvalidArgumentException(
-                    'Cannot use a nonexistent path or null value. ' .
-                    'Use a callback to provide default values if necessary.'
-                );
+                throw new InvalidArgumentException(sprintf(
+                    'Cannot group by path that does not exist or contains a null value. ' .
+                    'Use a callback to return a default value.',
+                    $path
```
Comment (markstory) [inline src/Collection/CollectionTrait.php:275]: Psalm isn't happy about this using sprintf() with no placeholders. Perhaps these should be basic strings?

Comment (othercorey) [reply src/Collection/CollectionTrait.php:275]: I was printing the path name before I remembered it can be a closure. Will remove it.

File: src/Collection/CollectionTrait.php [hunk 2]
```diff
@@ -270,8 +270,8 @@ public function groupBy($path): CollectionInterface
             $pathValue = $callback($value);
             if ($pathValue === null) {
                 throw new InvalidArgumentException(
-                    'Cannot use a nonexistent path or null value. ' .
-                    'Use a callback to provide default values if necessary.'
+                    'Cannot group by path that does not exist or contains a null value. ' .
+                    'Use a callback to return a default value.',
```
Comment (markstory) [inline src/Collection/CollectionTrait.php:274]:
```suggestion
                    'Use a callback to return a default value.'
```

File: src/Collection/CollectionTrait.php [hunk 3]
```diff
@@ -291,8 +291,8 @@ public function indexBy($path): CollectionInterface
             $pathValue = $callback($value);
             if ($pathValue === null) {
                 throw new InvalidArgumentException(
-                    'Cannot use a nonexistent path or null value. ' .
-                    'Use a callback to provide default values if necessary.'
+                    'Cannot index by path that does not exist or contains a null value. ' .
+                    'Use a callback to return a default value.',
```
Comment (markstory) [inline src/Collection/CollectionTrait.php:295]:
```suggestion
                    'Use a callback to return a default value.'
```
The trailing commas are an error in older php

Comment (othercorey) [reply src/Collection/CollectionTrait.php]: Apparently removing a parameter is very difficult work.
